conf.d/android-sdk.fish

```diff
@@ -1,13 +1,18 @@
-set paths $ANDROID_HOME/{tool?,platform-tool?}
-  or set paths $ANDROID_SDK_ROOT/{tool?,platform-tool?}
-  or set paths $HOME/Android/Sdk/{tool?,platform-tool?}
-
-if test (count $paths) -gt 0
-  set -l IFS /
-  echo $paths[1] | read -la components
-  set -e components[-1]
-  printf '/%s' $components | read -l root
-
-  set -gx PATH $PATH $paths
-  set -gx ANDROID_HOME $root
+# check for common Android SDK locations
+for root in $ANDROID_SDK_ROOT $ANDROID_SDK_ROOT $HOME/Android/Sdk $HOME/Library/Android/sdk
+  set -l paths
+  if test -d $root
+    # the SDK root exists as a directory so let's try to add the binary directories
+    for bin in $root/cmdline-tools/latest/bin "$root/build-tools/"(ls $root/build-tools)[-1] $root/platform-tools $root/emulator 
```

Comment:
These quotes are literally useless - unlike bash, fish does not split the variable after it has been set. Since $root is our for-loop variable, it's guaranteed to have exactly one element. So `$root/build-tools` without quotes always works. And if it didn't, the mistake would already have happened in the `if test -d $root` above!

The `ls`, which is how you're trying to get the `<version>` in `build-tools/<version>` should just be a `*` glob. Fish sorts globs, so that should work.

Reply:
I think you might need to reverse the sort-order, that's easy enough:

```fish
set -l build_tools $root/build-tools/*
for bin in ... $build_tools[-1..1] ...
```

Reply:
Strange, I didn't think it was working without the quotes, but now it is. Your suggestion adds all the build-tools versions, but I've made it index [-1] so we just get the most recent version.